scan/tls_session_scanner.go

```diff
@@ -0,0 +1,63 @@
+package scan
+
+import (
+	"crypto/tls"
+	"net"
+)
+
+// SessionResumptionFamily contains tests of host TLS Session Resumption
+var SessionResumptionFamily = &Family{
+	"SessionResumptionFamily",
+	"Scanners related to session resumption",
+	[]*Scanner{SessionResumptionScanner},
+}
+
+// SessionResumeScanner tests that host is able to resume sessions accross all addresses.
+var SessionResumptionScanner = &Scanner{
+	"SessionResumeScanner",
+	"Host is able to resume sessions accross all addresses.",
+	sessionResumeScan,
+}
+
+type sessionCache struct {
+	cache map[string][]*tls.ClientSessionState
+}
+
+func newSessionCache() *sessionCache {
+	return &sessionCache{make(map[string][]*tls.ClientSessionState)}
+}
+func (s *sessionCache) Get(sessionKey string) (mostRecent *tls.ClientSessionState, ok bool) {
+	states, ok := s.cache[sessionKey]
+	mostRecent = states[len(states)-1]
+	return
+}
+func (s *sessionCache) Put(sessionKey string, cs *tls.ClientSessionState) {
+	s.cache[sessionKey] = append(s.cache[sessionKey], cs)
+}
+func sessionResumeScan(host string) (grade Grade, output Output, err error) {
+	addrs, err := net.LookupHost(host)
+	if err != nil {
+		return
+	}
+	sessions := newSessionCache()
+	output = sessions
+	config := &tls.Config{ClientSessionCache: sessions}
+	for _, addr := range addrs {
+		var conn *tls.Conn
+		conn, err = tls.Dial(Network, addr, config)
```

Comment:
I recommend to use DialWithDialer, so we can control the timeout and easy to tweak. It's otherwise the system default timeout, which could be like 30 seconds.